web/src/portico/signup.ts

```diff
@@ -342,3 +343,102 @@ $(() => {
         }
     });
 });
+
+function get_or_set(
+    input_selector: string,
+    keep_leading_whitespace?: boolean,
+    no_trim?: boolean,
+): (newval?: string) => string {
+    return function () {
+        const $elem = $<HTMLInputElement | HTMLTextAreaElement>(input_selector);
+
+        if (!$elem.length) {
+            return "";
+        }
+
+        const oldval = $elem.val();
+        if (oldval === undefined) {
+            return "";
+        }
+
+        if (no_trim) {
+            return oldval;
+        } else if (keep_leading_whitespace) {
+            return oldval.trimEnd();
+        }
+        return oldval.trim();
+    };
+}
+
+let password_too_long = false;
+
+function set_password_too_long(status: boolean, is_signup: boolean): void {
```

Comment:
I think we can avoid some of these functions and just have the code directly in `check_overflow_password`.